### ISSUE TITLE
Unspecified value parameter jtagHeaderIgnoreWidth

### ISSUE THREAD

**Issue by FaizanAhmad626** [SpinalHDL/VexRiscv#332]
I am running into this error while generating core from VexRiscv.scala for ztachip. There is some new parameter "jtagHeaderIgnoreWidth" whose value is not declared in this file [https://github.com/ztachip/ztachip/blob/master/HW/examples/GHRD/MyVexRiscv.scala#L206](https://github.com/ztachip/ztachip/blob/master/HW/examples/GHRD/MyVexRiscv.scala#L206).
![image](https://user-images.githubusercontent.com/111587195/232323626-21acc0d9-1bd1-4cd4-b725-b444adce9978.png)

What to do about it?

**Comment by Dolu1990** [comment]
Hi,

Hmm you can do fromJtagInstructionCtrl(ClockDomain(tap.TCK), 0)

That should be good then.

**Comment by FaizanAhmad626** [comment]
Okay that issue is resolved but now I am having these errors:
![image](https://user-images.githubusercontent.com/111587195/233824688-493e8685-4ecf-4830-b2ac-d35ccbb521ef.png)

**Comment by ztachip** [comment]
I notice that my run has an older spinalHDL version v1.6.0 while yours is v1.8.1
Probably there is some new syntax on latest release.
I will need to upgrade to latest spinalHDL release to fix this.
Meanwhile, if you change the MaxPendingWrite in AXI crossbar, that should do the same.

________________________________

**Comment by FaizanAhmad626** [comment]
Okay.

**Comment by FaizanAhmad626** [comment]
Now I added 'C' extension to [MyVexRiscv.scala](https://github.com/ztachip/ztachip/blob/master/HW/examples/GHRD/MyVexRiscv.scala) by adding this line here:
![image](https://user-images.githubusercontent.com/111587195/235160875-2f03c5ab-0a92-4315-8063-1ac903dc714c.png)

Not sure if its correct or not. But its still giving those errors which @ztachip noticed is due to running newer version of SpinalHDL.

**Comment by Dolu1990** [comment]
What VexRiscv github branch / SpinalHDL is used ?

**Comment by FaizanAhmad626** [comment]
I am using SpinalHDL v 1.8.1 whereas this scala file [https://github.com/ztachip/ztachip/blob/master/HW/examples/GHRD/MyVexRiscv.scala](https://github.com/ztachip/ztachip/blob/master/HW/examples/GHRD/MyVexRiscv.scala) is using SpinalHDL v 1.6.0

**Comment by FaizanAhmad626** [comment]
> What VexRiscv github branch / SpinalHDL is used ?

I am using master branch for building VexRiscv from scala. But not sure about the branch of MyVexRiscv.

**Comment by Dolu1990** [comment]
Ahhh i think that the issue, master VexRiscv is meant to be used with a release of SpinalHDL, not dev.

So i think you can fix that jtagHeaderIgnoreWidth to 0, that should work fine.
Or switch to vexriscv dev aswell

**Comment by FaizanAhmad626** [comment]
I tried master version of SpinalHDL and didn't work.

Then I tried dev version of VexRiscv and still didn't work.

I may be making a mistake somewhere may be. I tried fullGen version and it compiled successfully but I don't know why its giving errors in MyVexRiscv version.

**Comment by ztachip** [comment]
Can you try the example scala that comes with VexRiscv
________________________________

**Comment by FaizanAhmad626** [comment]
I've tried this example:
sbt "runMain vexriscv.demo.GenFull"
And it generated the Verilog file successfully.

**Comment by ztachip** [comment]
Try copy MyVexRiscv.scala to same folder where you find GenFull.scala and try to compile again
Make sure there is just one copy of MyVexRiscv.scala in the VexRiscv folder since the compiler would look for all MyVexRiscv.scala from all folders under VexRiscv folder.
________________________________

**Comment by FaizanAhmad626** [comment]
I tried and again failed. I think there is some branch issue may be. Which branch did you use? I tried master branch and it failed to compile MyVexRiscv.scala. The dev branch even failed to compile GenFull.scala I don't know why.

![image](https://user-images.githubusercontent.com/111587195/235888099-8f521c32-691c-460c-92a8-0f02b0e2b83b.png)
And by looking at these errors I concluded that there is some issue related to the libraries you imported for MyVexRiscv.scala which are not imported in the GenFull.scala file. This is due to the extra functionalities in MyVexRiscv version so may be some extra per-requisites must also be required. Right now I am just cloning VexRiscv from GitHub and installed its per-requisites only.

I don't know if this will be emailed to you or not so I am mentioning you in this comment @ztachip
